### PR TITLE
The dhcpv6 parser

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

The dhcpv6 parser
## Impacts

New code does not effect other code
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
## Delete branch after merge

NO

### CODE REVIEW

File: t/dhcpv6.t [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+use strict;
+use warnings;
+use lib qw(/usr/local/pf/lib);
+use pf::util::dhcpv6;
+use Net::Pcap qw(pcap_open_offline pcap_loop);
+use bytes;
+
+my $filename = '/usr/local/pf/t/data/dhcpv6cap.pcapng';
+#my $filename = '/usr/local/pf/t/data/dhcpv6-sample-2.pcap';
```
Comment (fdurand) [inline t/dhcpv6.t]: Remove

File: t/dhcpv6.t [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+my $err      = '';
+
+my $pcap = pcap_open_offline($filename, \$err)
+  or die "Can't read '$filename': $err\n";
+
+our $count = 1;
+my $data = \$count;
+use Data::Dumper;
```
Comment (fdurand) [inline t/dhcpv6.t]: Remove

File: t/dhcpv6.t [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+my $value = pcap_loop($pcap, -1, \&process_packet, $data);
+
+sub process_packet {
+    my ($user_data, $header, $packet) = @_;
+    my ($eth_obj, $ip_obj, $udp_obj, $dhcp) = decompose_dhcpv6($packet);
+    $dhcp->{count} = $$user_data;
+    $$user_data++;
+#    print Dumper $dhcp;
```
Comment (fdurand) [inline t/dhcpv6.t]: Remove

**Comment by fdurand** [comment]
The lib looks ok for me

**Comment by jrouzierinverse** [comment]
Please review

**Comment by fdurand** [comment]
Do we have to keep dhcpv6-sample-1.pcap and dhcpv6-sample-2.pcap in t/ directory ? only the third one is use for the test.

**Comment by jrouzierinverse** [comment]
I would like to keep them since they are have additional options that are not in the main one.

**Comment by jrouzierinverse** [comment]
Please review

**Comment by louismunro** [comment]
LGTM. Merging.

**Pull request merged**